src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs

```diff
@@ -1148,6 +1148,61 @@ impl ToQemuParams for Rtc {
     }
 }
 
+#[derive(Debug)]
+struct RngDevice {
+    // id is the device ID
+    id: String,
+
+    // filename is the entropy source on the host
+    filename: String,
+
+    // max_bytes is the bytes allowed to guest to get from the host’s entropy per period
+    max_bytes: u32,
+
+    // period is duration of a read period in seconds
+    period: u32,
+
+    // transport is the virtio transport for this device.
+    transport: String,
+}
+
+impl RngDevice {
+    fn new() -> RngDevice {
+        RngDevice {
+            id: "rng0".to_owned(),
+            filename: "/dev/urandom".to_owned(),
+            max_bytes: 1024,
+            period: 1000,
+            transport: "virtio-rng-pci".to_owned(),
+        }
+    }
+}
+
+#[async_trait]
+impl ToQemuParams for RngDevice {
+    async fn qemu_params(&self) -> Result<Vec<String>> {
+        let mut object_params = Vec::new();
+        let mut device_params = Vec::new();
```

Comment:
qemu-rs uses a struct per command line switch for code readability.  `RngDevice` should be split into a `DeviceRng` part to generate `-device virtio-rng...` and an `ObjectRngRandom` to generate the `-object rng-random` part.